dpc/src/testnet1/outer_circuit/outer_circuit_gadget.rs

```diff
@@ -299,12 +289,12 @@ where
     // Verify the InnerSNARK proof
     // ************************************************************************
 
-    let inner_snark_vk = <C::InnerSNARKGadget as SNARKVerifierGadget<_, _, _>>::VerificationKeyGadget::alloc(
+    let inner_snark_vk = <C::InnerSNARKGadget as SNARKVerifierGadget<_>>::VerificationKeyGadget::alloc(
```

Comment:
Now the `SNARKVerifierGadget` only needs to pass one type parameter, since it can now infer the fields from the `SNARK` trait implementation.